### PR TITLE
Update changeset format

### PR DESCRIPTION
**Pull request by hatched**
The bundlelib changeset format was modified so these modifications are now supported in the GUI. The GUI supports both the new and old format although the old format will be deprecated soon.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

- Use the following code to create a file called 'changes.json' Drag the following file to the canvas (drop it on the onboarding elements (it will just open the file if you drop anywhere else)) and it should import it. (new format)
- Refresh
- Deploy a bundle using deploy-target and it should deploy the bundle. (old format)

```
[
{"args": ["cs:trusty/hdp-hadoop-8"], "requires": [], "id": "addCharm-0", "method": "addCharm"}
,
{"args": ["$addCharm-0", "compute-node", {}, ""], "requires": ["addCharm-0"], "id": "deploy-1", "method": "deploy"}
,
{"args": ["$deploy-1", "service", {"gui-y": "997.2758247589375", "gui-x": "656.6386478556842"}], "requires": ["deploy-1"], "id": "setAnnotations-2", "method": "setAnnotations"}
,
{"args": ["cs:trusty/hdp-pig-4"], "requires": [], "id": "addCharm-3", "method": "addCharm"}
,
{"args": ["$addCharm-3", "hdp-pig", {}, ""], "requires": ["addCharm-3"], "id": "deploy-4", "method": "deploy"}
,
{"args": ["$deploy-4", "service", {"gui-y": "279.7241752410624", "gui-x": "675.9159435670526"}], "requires": ["deploy-4"], "id": "setAnnotations-5", "method": "setAnnotations"}
,
{"args": ["$addCharm-0", "yarn-hdfs-master", {}, ""], "requires": ["addCharm-0"], "id": "deploy-6", "method": "deploy"}
,
{"args": ["$deploy-6", "service", {"gui-y": "654.7241752410625", "gui-x": "657.3613521443158"}], "requires": ["deploy-6"], "id": "setAnnotations-7", "method": "setAnnotations"}
,
{"args": ["$deploy-4:namenode", "$deploy-6:namenode"], "requires": ["deploy-4", "deploy-6"], "id": "addRelation-8", "method": "addRelation"}
,
{"args": ["$deploy-6:namenode", "$deploy-1:datanode"], "requires": ["deploy-6", "deploy-1"], "id": "addRelation-9", "method": "addRelation"}
,
{"args": ["$deploy-6:resourcemanager", "$deploy-1:nodemanager"], "requires": ["deploy-6", "deploy-1"], "id": "addRelation-10", "method": "addRelation"}
,
{"args": ["$deploy-4:resourcemanager", "$deploy-6:resourcemanager"], "requires": ["deploy-4", "deploy-6"], "id": "addRelation-11", "method": "addRelation"}
,
{"args": ["$deploy-4:hadoop-nodes", "$deploy-1:hadoop-nodes"], "requires": ["deploy-4", "deploy-1"], "id": "addRelation-12", "method": "addRelation"}
,
{"args": ["$deploy-1", null], "requires": ["deploy-1"], "id": "addUnit-13", "method": "addUnit"}
,
{"args": ["$deploy-4", null], "requires": ["deploy-4"], "id": "addUnit-14", "method": "addUnit"}
,
{"args": ["$deploy-6", null], "requires": ["deploy-6"], "id": "addUnit-15", "method": "addUnit"}
]
```

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3221/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3223/
Test PASSed.

**Comment by huwshimi** [comment]
QA OK :+1:

File: jujugui/static/gui/src/app/utils/bundle-importer.js [hunk 1]
```diff
@@ -422,7 +437,16 @@ YUI.add('bundle-importer', function(Y) {
           ghostService.set('displayName', displayName);
 
           ghostService.set('config', config);
-
+          // Convert the string space delimeted constraints into an object
+          // as required by the juju api.
+          var constraints = {};
```
Comment (frankban) [inline jujugui/static/gui/src/app/utils/bundle-importer.js]: I think this is already done by the env (see env.prepareConstraints) so not sure if it's worth redoing it now.
In general, I think it should be safe to pass both string and object formats to deploy().

File: jujugui/static/gui/src/app/utils/bundle-importer.js [hunk 2]
```diff
@@ -505,9 +529,22 @@ YUI.add('bundle-importer', function(Y) {
...
+      var serviceId, charmUrl, size, name;
+      // The bundlelib no longer returns the same format as required by the
+      // GUI's add_unit env method so this re-maps the arguments so that they
+      // match.
+      // If the length is longer than 2 then it's still using the old format
+      // which includes the matching args.
+      // XXX This can be removed once the bundlelib has been updated across
+      // all platforms.
+      if (record.args.length === 2) {
+        // Move the machine placement to the third argument spot.
```
Comment (frankban) [inline jujugui/static/gui/src/app/utils/bundle-importer.js]: Thanks for these comments.

**Comment by frankban** [comment]
:+1: with a comment. This looks good Jeff thank you!

File: jujugui/static/gui/src/app/utils/bundle-importer.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/utils/bundle-importer.js]: Oops, removed that.

**Comment by hatched** [comment]
Thanks for the reviews @huwshimi @frankban 
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3238/
Test FAILed.

**Comment by hatched** [comment]
spurious ci failure

**Pull request merged**